http/delete_handler.go

```diff
@@ -139,24 +142,71 @@ func (h *DeleteHandler) handleDelete(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
-func decodeDeleteRequest(ctx context.Context, r *http.Request, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) (*deleteRequest, error) {
+func decodeDeleteRequest(ctx context.Context, r *http.Request, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) (*deleteRequest, influxql.Expr, error) {
+	jsonErr := &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "invalid request; error parsing request json",
+	}
+
 	dr := new(deleteRequest)
-	err := json.NewDecoder(r.Body).Decode(dr)
+	buf, err := io.ReadAll(r.Body)
```

Comment:
Why the shift to a buffered read here?

Reply:
One rule of thumb I use is that every error cause should return a different error. This enormously helps narrow down debugging searches.  I would use `fmt.Errorf("$s : %w", "What I was doing", jsonErr)` for each return of `jsonErr` in this function. `errors.Is()` will still show it is a `jsonErr`, but you will know if it is an error from `io.ReadAll` or `Decode` or `Unmarshal`

Reply:
And that will help us when we see this from a misbehaving customer and have to figure it out.